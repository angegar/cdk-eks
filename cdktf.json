{
  "language": "typescript",
  "app": "npx ts-node main.ts",
  "projectId": "6e6cbd44-4696-42c7-b82d-bb0b71d97f08",
  "terraformProviders": ["aws@~> 2.0"],
  "terraformModules": [],
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
