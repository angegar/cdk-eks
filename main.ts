// import { Construct } from "constructs";
import { App } from "cdktf";
// import * as aws from '@cdktf/provider-aws';
import { clusterConfig } from './config';
// import { TerraformProvider} from './provider';
import { EksStack } from "./eksstack";

const app = new App();

let clusterName: string = `${clusterConfig.application.environment}-${clusterConfig.application.project}`
let stack1: EksStack =new EksStack(app, clusterName, "1.23")
stack1.CreateNodeGroup(
	"intel",
	1,
	["t3.medium"],
	"x86_64")
	
stack1.CreateHostedZone("pe-demo.manomano.tech")

app.synth();