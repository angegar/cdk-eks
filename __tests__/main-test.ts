import "cdktf/lib/testing/adapters/jest"; // Load types for expect matchers
import { Testing } from "cdktf";
import * as aws from '@cdktf/provider-aws';

import { EksInfraGlobalStack } from '../main'


describe("My CDKTF Application", () => {

  // // All Unit testst test the synthesised terraform code, it does not create real-world resources
  describe("Global stack Unit testing using assertions", () => {
    it("should contain a role", () => {
      const app = Testing.app();
      const stack = new EksInfraGlobalStack(app, "stack", {
        environment: 'env',
        project: 'test',
        region: 'eu-west-3',
        type: 'infra'
      });

      expect(Testing.synth(stack)).toHaveResource(aws.iam.IamRole);
    });

    it("should contain a role with props", () => {
      const app = Testing.app();
      const stack = new EksInfraGlobalStack(app, "stack", {
        environment: 'env',
        project: 'test',
        region: 'eu-west-3',
        type: 'infra'
      });

      expect(Testing.synth(stack)).toHaveResourceWithProperties(aws.iam.IamRole, { name: "role-eks-cluster" });

    });
  });

  describe("Unit testing using snapshots", () => {
    it("should be valid", () => {
      const app = Testing.app();
      const stack = new EksInfraGlobalStack(app, "stack", {
        environment: 'env',
        project: 'test',
        region: 'eu-west-3',
        type: 'infra'
      });

      expect(Testing.synth(stack)).toMatchSnapshot();
    });
  });

});
