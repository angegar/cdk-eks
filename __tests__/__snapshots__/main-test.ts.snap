// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My CDKTF Application Unit testing using snapshots should be valid 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"profile\\": \\"manomano-env\\",
        \\"region\\": \\"eu-west-3\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_iam_role\\": {
      \\"eks-cluster-role\\": {
        \\"assume_role_policy\\": \\"{\\\\\\"Version\\\\\\":\\\\\\"2012-10-17\\\\\\",\\\\\\"Statement\\\\\\":[{\\\\\\"Effect\\\\\\":\\\\\\"Allow\\\\\\",\\\\\\"Principal\\\\\\":{\\\\\\"Service\\\\\\":\\\\\\"eks.amazonaws.com\\\\\\"},\\\\\\"Action\\\\\\":\\\\\\"sts:AssumeRole\\\\\\"}]}\\",
        \\"inline_policy\\": [
          {
            \\"name\\": \\"AmazonEKSClusterCloudWatchMetricsPolicy\\",
            \\"policy\\": \\"{\\\\\\"Version\\\\\\":\\\\\\"2012-10-17\\\\\\",\\\\\\"Statement\\\\\\":[{\\\\\\"Action\\\\\\":[\\\\\\"cloudwatch:PutMetricData\\\\\\"],\\\\\\"Resource\\\\\\":\\\\\\"*\\\\\\",\\\\\\"Effect\\\\\\":\\\\\\"Allow\\\\\\"}]}\\"
          }
        ],
        \\"managed_policy_arns\\": [
          \\"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy\\",
          \\"arn:aws:iam::aws:policy/AmazonEKSVPCResourceController\\"
        ],
        \\"name\\": \\"role-eks-cluster\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"s3\\": {
        \\"bucket\\": \\"mm-terraform-remote-states-global\\",
        \\"dynamodb_table\\": \\"tf-states-lock-table\\",
        \\"encrypt\\": true,
        \\"key\\": \\"env/infra/test-infra-global.tfstate\\",
        \\"profile\\": \\"manomano-support\\",
        \\"region\\": \\"eu-west-1\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"4.15.1\\"
      }
    }
  }
}"
`;
